internal/models/linking.go

```diff
@@ -49,7 +50,19 @@ type AccountLinkingResult struct {
 // - It's not possible to decide due to data inconsistency (MultipleAccounts) and the caller should decide
 //
 // Errors signal failure in processing only, like database access errors.
-func DetermineAccountLinking(tx *storage.Connection, provider, sub string, emails []string) (AccountLinkingResult, error) {
+func DetermineAccountLinking(tx *storage.Connection, config *conf.GlobalConfiguration, emails []p.Email, aud, provider, sub string) (AccountLinkingResult, error) {
```

Comment:
Maybe rename `provider` to `providerType` or `providerName` to avoid the conflict from the import?